x-pack/plugins/observability/public/plugin.ts

```diff
@@ -172,13 +176,37 @@ export class Plugin
     }
 
     this.navigationRegistry.registerSections(
-      of([
-        {
-          label: '',
-          sortKey: 100,
-          entries: [{ label: 'Overview', app: 'observability-overview', path: '/overview' }],
-        },
-      ])
+      from(updater$).pipe(
+        map((value) => {
+          const deepLinks = value(app)?.deepLinks ?? [];
+
+          const overviewLink = {
+            label: i18n.translate('xpack.observability.overviewLinkTitle', {
+              defaultMessage: 'Overview',
+            }),
+            app: observabilityAppId,
+            path: '/overview',
+          };
+
+          // Reformat the visible links to be NavigationEntry objects instead of
+          // AppDeepLink objects.
+          const otherLinks: NavigationEntry[] = deepLinks
+            .filter((link) => link.navLinkStatus === AppNavLinkStatus.visible)
+            .map((link) => ({
+              app: observabilityAppId,
+              label: link.title,
+              path: link.path ?? '',
+            }));
+
+          return [
```

Comment:
I wouldn't mind assigning this to `const sections = [ ... ]` and then returning it, as it's pretty far removed from `registerSections`. Or the observable itself is called something like `sections$`.